jdaviz/configs/cubeviz/plugins/tests/test_cubeviz_aperphot.py

```diff
@@ -253,90 +258,115 @@ def _compare_table_units(orig_tab, new_tab, orig_flux_unit=None,
 
-    # make sure inputs were re-computed in new units
-    # after the unit change
-    assert_allclose(ap.flux_scaling, 3631)
-    assert_allclose(ap.background_value, 4.9e7)
+        if new_flux_unit != flux_unit_str:  # dont compare same units
 
-    # re-do photometry and make sure table is in new units
-    # and consists of the same results as before converting units
-    ap.vue_do_aper_phot()
-    new_tab = Table(ap.results)
+            new_un = u.Unit(new_flux_unit)
+            # reset to 'original' unit so were not comparing to the last iteration
+            uc.flux_unit.selected = flux_unit_str
+            ap.background_selected == 'Manual'
+            ap.background_value = 1.
+            ap.flux_scaling = 1.
 
-    _compare_table_units(orig_tab, new_tab, orig_flux_unit=u.MJy,
-                         new_flux_unit=u.Jy)
+            # then set to new unit
+            uc.flux_unit.selected = new_flux_unit
 
-    # test manual background and flux scaling option input in current
-    # units (Jy / sr) will be used correctly and converted to data units
-    ap.background_selected == 'Manual'
-    ap.background_value = 1.0e7
-    ap.flux_scaling = 1000
-    ap.vue_do_aper_phot()
-    orig_tab = Table(ap.results)
+            # make sure display units in aperture phot plugin reflect change
+            # multiply out angle to avoid formatting inconsistencies
+            assert (u.Unit(ap.display_unit) * angle_unit) == (u.Unit(new_flux_unit))
+            assert ap.flux_scaling_display_unit == new_flux_unit
 
-    # change units back to MJy/sr from Jy/sr
-    uc.flux_unit.selected = 'MJy'
+            # make sure background and flux scaling were converted to new unit
+            assert_allclose((ap.background_value * new_un).to(flux_unit, equiv).value, 1.)
+            assert_allclose((ap.flux_scaling * new_un).to(flux_unit, equiv).value, 1.)
 
-    # make sure background input in Jy/sr is now in MJy/sr
-    assert_allclose(ap.background_value, 10)
-    assert_allclose(ap.flux_scaling, 0.001)
+            ap.vue_do_aper_phot()
+            new_tab = Table(ap.results)
 
-    # and that photometry results match those before unit converson,
-    # but with units converted
-    ap.vue_do_aper_phot()
-    new_tab = Table(ap.results)
+            # if ap. phot silently fails, then 'new_tab' will just be the last
+            # calculated one, so make sure this didn't happen
+            assert not np.all(orig_tab == new_tab)
+
+            # compare output tables row by row between original and new unit
+            _compare_table_units(orig_tab, new_tab, orig_flux_unit=flux_unit,
+                                 new_flux_unit=u.Unit(new_flux_unit),
+                                 equivalencies=equiv)
 
-    _compare_table_units(orig_tab, new_tab, orig_flux_unit=u.Jy,
-                         new_flux_unit=u.MJy)
+            # todo: figure out how to test radial profile and curve of growth plots
```

Comment:
Is this TODO going to be in this PR or a different ticket? If different, please link here. Thanks.

Reply:
not in this, i need to make a jira ticket ill link once i do that